api-service/src/controllers/DatasetUpdate/DatasetUpdate.ts

```diff
@@ -0,0 +1,297 @@
+import httpStatus from "http-status";
+import { Request, Response } from "express";
+import { ResponseHandler } from "../../helpers/ResponseHandler";
+import { schemaValidation } from "../../services/ValidationService";
+import DatasetUpdate from "./DatasetUpdateValidationSchema.json";
+import _ from "lodash";
+import { DatasetStatus } from "../../types/DatasetModels";
+import { ErrorObject } from "../../types/ResponseModel";
+import { DatasetDraft } from "../../models/DatasetDraft";
+import logger from "../../logger";
+import { defaultDatasetConfig } from "../../configs/DatasetConfigDefault";
+import { DatasetTransformationsDraft } from "../../models/TransformationDraft";
+import moment from "moment";
+
+const datasetUpdate = async (req: Request, res: Response) => {
+    try {
+        const datasetBody = req.body;
+        const isRequestValid: Record<string, any> = schemaValidation(datasetBody, DatasetUpdate)
+        if (!isRequestValid.isValid) {
+            return ResponseHandler.errorResponse({
+                message: isRequestValid.message,
+                statusCode: 400,
+                errCode: "BAD_REQUEST"
+            } as ErrorObject, req, res);
+        }
+
+        const { dataset_id, ...rest } = datasetBody
+        if (_.isEmpty(rest)) {
+            return ResponseHandler.errorResponse({
+                message: "Invalid request body",
```

Comment:
Message should be proper says what is not correct
`Atleast one field is required to updated dataset other than dataset_id`